compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6782,12 +6782,16 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
             requiredParams.add(nonRestParam);
         }
 
+        List<String> includedRecordParamNames = new ArrayList<>();
```

Comment:
Shall we try to populate data structures with the initial sizes? This will improve performance. I believe java already has set this to 16 which is not accurate for our code. It is fine if it is not the exact amount but setting it to the original list size will prevent using extra memory and processor usage shift elements in arrays,

Reply:
Changed.